### PR TITLE
REGISTRY-3183  Fix search query appear in searchbox with result page

### PR DESCRIPTION
**Pull request by tmkasun**

### CODE REVIEW

File: apps/store/extensions/app/store-common/pages/top_assets.jag [hunk 1]
```diff
@@ -65,8 +66,9 @@ require('/modules/store.js').exec(function (ctx) {
         }
     }
     var qtemp = ctx.request.getParameter('q') || '';
-    var qjson = parse('{'+qtemp+'}');
-    output.searchQuery = qjson.name||'';
+    var searchQuery = assetApi.formatSearchQuery(qtemp);
+    output.searchQuery = searchQuery||'';
```
Comment (mushthaq33) [inline apps/store/extensions/app/store-common/pages/top_assets.jag]: Formatting issue

File: apps/store/extensions/assets/default/pages/list.jag [hunk 2]
```diff
@@ -153,7 +153,7 @@ require('/modules/store.js').exec(function(ctx) {
     var output = am.render(assets, page).list();
     output.paging = assetApi.assetsPaging(assets, type, ctx.request);
     var qtemp = ctx.request.getParameter('q') || '';
-    var qjson = parse('{'+qtemp+'}');
-    output.searchQuery = qjson.name||'';
+    var searchQuery = assetApi.formatSearchQuery(qtemp);
+    output.searchQuery = searchQuery||'';
```
Comment (mushthaq33) [inline apps/store/extensions/assets/default/pages/list.jag]: Formatting issue

File: apps/store/modules/asset-api.js [hunk 3]
```diff
@@ -540,4 +540,24 @@ var responseProcessor = require('utils').response;
...
+    
+    api.formatSearchQuery = function (queryString) {
+        var searchQuery = "";
+        var qjson = parse('{' + queryString + '}');
+
+        var searchKeys = Object.keys(qjson);
+        if ((searchKeys.length === 1) && (searchKeys.indexOf("name") >= 0)) {
+            searchQuery += qjson[searchKeys.pop()];
+        }
+        else {
```
Comment (splinter) [inline apps/store/modules/asset-api.js]: Move to line 551 so that it appears as  }  else {

Comment (tmkasun) [inline apps/store/modules/asset-api.js]: Thanks @splinter , This method *formatSearchQuery* has been moved to *request.js* from [this commit](https://github.com/wso2/carbon-store/commit/8e2edd3139a54ee3b1571eacabbffcc7ccad25d3#diff-26fa7e5ae96217c017cd62b4d21baee7), but this formatting issue was still there. I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/1cfd4a532c0b6fcbb158ab5da98f004afac2612c)

File: apps/store/extensions/assets/default/pages/list.jag (hunk 2 above)
Comment (tmkasun) [inline apps/store/extensions/assets/default/pages/list.jag]: Thanks @mushthaq33 , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/1cfd4a532c0b6fcbb158ab5da98f004afac2612c)

File: apps/store/extensions/app/store-common/pages/top_assets.jag (hunk 1 above)
Comment (tmkasun) [inline apps/store/extensions/app/store-common/pages/top_assets.jag]: Thanks @mushthaq33 , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/1cfd4a532c0b6fcbb158ab5da98f004afac2612c)

File: apps/store/modules/asset-api.js [hunk 4]
```diff
@@ -540,4 +540,24 @@ var responseProcessor = require('utils').response;
...
+            for (var keyIndex in searchKeys) {
+                var key = searchKeys[keyIndex];
+                var value = qjson[key];
+                searchQuery += key + ":" + value + " ";
+            }
+        }
+        searchQuery = searchQuery.trim();
+        return searchQuery;
+    };
+
```
Comment (thushara35) [inline apps/store/modules/asset-api.js]: Even though there is no difference between using single quotes over double quotes, try to maintain the consistency throughout the code.  L546 you have used ''

Comment (tmkasun) [inline apps/store/modules/asset-api.js]: Thanks @thushara35 , I have changed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/993a3cd2fad8bce67dd7caca21bcc51675b3d9e5)

**Pull request merged**